src/mpyl/steps/deploy/k8s/__init__.py

```diff
@@ -23,6 +26,35 @@ def get_namespace(run_properties: RunProperties, project: Project) -> Optional[s
     return get_namespace_from_project(project)
 
 
+def rollout_restart_deployment(logger: Logger, namespace: str, deployment: str):
+    # from https://stackoverflow.com/a/67491253
+    v1_apps = client.AppsV1Api()
+
+    now = datetime.datetime.utcnow()
+    now_str = now.isoformat("T") + "Z"
+    body = {
+        "spec": {
+            "template": {
+                "metadata": {
+                    "annotations": {"kubectl.kubernetes.io/restartedAt": now_str}
+                }
+            }
+        }
+    }
+    try:
+        logger.info(f"Starting rollout restart of {deployment}...")
+        _, status_code, _ = v1_apps.patch_namespaced_deployment_with_http_info(
+            deployment, namespace, body, pretty="true"
+        )
+        logger.info(
+            f"Rollout restart of {deployment} finished with statuscode {status_code}"
+        )
+    except ApiException as api_exception:
+        logger.info(
+            f"Exception when calling AppsV1Api->read_namespaced_deployment_status: {api_exception}\n"
+        )
+
```

Comment:
Shouldn't we signal failure here so we can fail the execute step?

Reply:
yes indeed 👍

Reply:
like discussed i am returning `Output`s in these function that can get evaluated by the DagsterDeploy step (and exiting early if failure)